pallets/loans/src/loan.rs

```diff
@@ -207,9 +205,8 @@ impl<T: Config> Pallet<T> {
             return Err(Error::<T>::OracleCurrencyPriceNotReady);
         }
 
-        (collateral_factor * collateral)
-            .checked_mul(exchange_rate)
-            .and_then(|r| r.checked_div(RATE_DECIMAL))
+        exchange_rate
+            .checked_mul_int(collateral_factor * collateral)
```

Comment:
looks great, 
I have some concerns here, seems like it will return a fixed-point type, but this method will return the Balance type, if this is ok here?